build/configs/esp32_DevKitC/Make.defs

```diff
@@ -143,7 +150,10 @@ OBJEXT = .o
 LIBEXT = .a
 EXEEXT = .elf
 
-LDFLAGS += -nostartfiles -nodefaultlibs --gc-sections
+LDFLAGS += -nostartfiles -nodefaultlibs 
+ifneq ($(CONFIG_ENTRY_HELLO),y)
+LDFLAGS += -gc-sections
+endif
```

Comment:
I don't know this is necessary. Why do we need to add some code with example config?

Reply:
this flag can be added without example config as well. I am changing it